### PR TITLE
docs(app-vite&app-webpack): Accessing router in Pinia&Vuex

### PR DESCRIPTION
**Pull request by rashidpathiyil**
### Documented accessing router in stores

**What kind of change does this PR introduce?** 

- [x] Documentation

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

Ref #14068

### CODE REVIEW

File: docs/src/pages/quasar-cli-vite/state-management-with-pinia.md [hunk 1]
```diff
@@ -148,3 +148,24 @@ export default {
 ```
 
 [More info on defining a Pinia store](https://pinia.vuejs.org/core-concepts/).
+
+## Accessing router in Pinia store
+Simply use `this.router` in Pinia stores to get access to the router.
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-vite/state-management-with-pinia.md]: Let's leave an empty line after the title for readability(_codebase-wise_) and to match with the other usages.

Let's update the title slightly.

```suggestion
## Accessing the router in Pinia stores

Simply use `this.router` in Pinia stores to get access to the router.
```

File: docs/src/pages/quasar-cli-vite/state-management-with-pinia.md [hunk 2]
```diff
@@ -148,3 +148,24 @@ export default {
...
+
+```js
+this.router.push('...')
+```
+
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-vite/state-management-with-pinia.md]: This is not clear enough and redundant as you have a better example below that.
```suggestion
```

File: docs/src/pages/quasar-cli-vite/state-management-with-vuex.md [hunk 3]
```diff
@@ -455,3 +455,17 @@ export default defineComponent({
 });
 </script>
 ```
+
+## Accessing router in Vuex store
+Simply use `this.$router` in Vuex stores to get access to the router.
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-vite/state-management-with-vuex.md]:
```suggestion
## Accessing the router in Vuex stores

Simply use `this.$router` in Vuex stores to get access to the router.
```

File: docs/src/pages/quasar-cli-webpack/state-management-with-pinia.md [hunk 4]
```diff
@@ -148,3 +148,24 @@ export default {
 ```
 
 More info on defining a Pinia store [here](https://pinia.vuejs.org/core-concepts/).
+
+## Accessing router in Pinia store
+Simply use `this.router` in Pinia stores to get access to the router.
+
+```js
+this.router.push('...')
+```
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-webpack/state-management-with-pinia.md]: Combined suggestion from the Vite page above.
```suggestion
## Accessing the router in Pinia stores

Simply use `this.router` in Pinia stores to get access to the router.
```

File: docs/src/pages/quasar-cli-webpack/state-management-with-vuex.md [hunk 5]
```diff
@@ -455,3 +455,17 @@ export default defineComponent({
 });
 </script>
 ```
+
+## Accessing router in Vuex store
+Simply use `this.$router` in Vuex stores to get access to the router.
+
+```js
+this.$router.push('...')
+```
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-webpack/state-management-with-vuex.md]: Combined suggestion from the Vite page above.
```suggestion
## Accessing the router in Vuex stores

Simply use `this.$router` in Vuex stores to get access to the router.
```

**Review by yusufkandemir — CHANGES_REQUESTED** [review]
Nice 👏 Finally, some documentation for this feature! Made some small suggestions.

**Comment by rashidpathiyil** [comment]
> Nice 👏 Finally, some documentation for this feature! Made some small suggestions.

✅ Done @yusufkandemir

**Review by yusufkandemir — APPROVED** [review]

File: docs/src/pages/quasar-cli-vite/state-management-with-pinia.md [hunk 6]
```diff
@@ -148,3 +148,21 @@ export default {
 ```
 
 [More info on defining a Pinia store](https://pinia.vuejs.org/core-concepts/).
+
+## Accessing the router in Pinia stores
+
+Simply use `this.router` in Pinia stores to get access to the router.
+
+See an example of accessing router in Pinia store action:
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/state-management-with-pinia.md]: I'd say "Here is an example" here or just remove this line as is on the other pages.

File: docs/src/pages/quasar-cli-vite/state-management-with-vuex.md [hunk 7]
```diff
@@ -455,3 +455,18 @@ export default defineComponent({
...
+
+## Accessing the router in Vuex stores
+
+Simply use `this.$router` in Vuex stores to get access to the router.
+
+```js
+this.$router.push('...')
+```
+
+See an example of accessing router in Vuex store:
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/state-management-with-vuex.md]: Same here. Use "This is an example....".... or "Here is an example....".

File: docs/src/pages/quasar-cli-webpack/state-management-with-pinia.md [hunk 8]
```diff
@@ -148,3 +148,21 @@ export default {
 ```
 
 More info on defining a Pinia store [here](https://pinia.vuejs.org/core-concepts/).
+
+## Accessing the router in Pinia stores
+
+Simply use `this.router` in Pinia stores to get access to the router.
+
+See an example of accessing router in Pinia store action:
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-webpack/state-management-with-pinia.md]: And again.

File: docs/src/pages/quasar-cli-webpack/state-management-with-vuex.md [hunk 9]
```diff
@@ -455,3 +455,14 @@ export default defineComponent({
 });
 </script>
 ```
+
+## Accessing the router in Vuex stores
+
+Simply use `this.$router` in Vuex stores to get access to the router.
+
+See an example of accessing router in Vuex store:
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-webpack/state-management-with-vuex.md]: Last one.

**Review by smolinari — CHANGES_REQUESTED** [review]
If you could, please update the "See an example..." sentences to the suggested changes. 

Thanks for your contribution!!! :+1:

**Review by rashidpathiyil — COMMENTED** [review]
Updated sentences as per your suggestion. @smolinari

**Review by smolinari — APPROVED** [review]

File: docs/src/pages/quasar-cli-vite/state-management-with-vuex.md [hunk 10]
```diff
@@ -455,3 +455,18 @@ export default defineComponent({
...
 ```
+
+## Accessing the router in Vuex stores
+
+Simply use `this.$router` in Vuex stores to get access to the router.
+
+```js
+this.$router.push('...')
+```
+
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-vite/state-management-with-vuex.md]:
```suggestion
```

**Pull request merged**